packages/ast-spec/tsconfig.json

```diff
@@ -1,10 +1,15 @@
 {
-  "extends": "./tsconfig.build.json",
+  "extends": "../../tsconfig.base.json",
   "compilerOptions": {
-    "composite": false,
-    "rootDir": "."
+    "outDir": "../../dist/out-tsc/ast-spec"
   },
-  "include": ["src", "typings", "tests", "tools", "**/fixtures/**/config.ts"],
-  "exclude": ["**/fixtures/**/fixture.ts", "**/fixtures/**/fixture.tsx"],
-  "references": [{ "path": "../typescript-estree/tsconfig.build.json" }]
+  "files": [],
+  "references": [
+    {
+      "path": "./tsconfig.build.json"
+    },
+    {
+      "path": "./tsconfig.spec.json"
+    }
+  ]
```

Comment:
This is interesting; I assume these are the main code and the tests, in which case I would have actually expected you to have `tsconfig.json` here be the main code, then have a `tsconfig.test.json` (or `spec`, that's bikeshedding) separate and then referenced from the repo root

Reply:
The idea behind this approach/config structure was as follows:

- A `tsconfig.json` for a project will power its type-checking. This is is not expected to produce any .js etc artifacts. It is logically expected to type-check all files, both production and test files.
- An optional `tsconfig.lib.json` (customizable name, in this workspace the convention is `tsconfig.build.json`) would be used to control a `tsc` powered build, in which .js etc artifacts _are_ produced. It should naturally not include any test files as part of its build process.
  - The presence of this file is also how Nx, via the new plugin, can infer that a build "target" should exist for this project (meaning you can run `nx build ast-spec` and it just works, including distributable caching and other core Nx features, without any specific Nx config at the project level. 
- An optional `tsconfig.spec.json` seemed to make sense as a corollary of the `tsconfig.lib.json`/`tsconfig.build.json`, but note that its presence is not used to infer any targets. It is only there to pick up all the files that the build config would exclude and possible be passed to a test runner like jest with ts-jest as config.

Does that all make sense? Is there a different/more optimal way to achieve the above objectives around having typecheck and build be separate actions/targets?

Reply:
That does make some sense, I was jus tlooking for ways to simplify things. But this is not dissimilar to how vite does things where it makes an extra tsconfig that's referenced just for config files.